### ISSUE TITLE
Check the Roslyn dynamic file processing based on the trace to look for improvements

### ISSUE THREAD

**Issue by mikadumont** [dotnet/vscode-csharp#5755]
\\mlangfs1\Public\dawengie\OrchardCodeWith111.zip

Anecdotally, from looking at the C# and Razor log output in VS Code: 

- Razor extension bits startup very quick, just spinning through each dynamic document
- Roslyn spends a lot longer processing dynamic files after Razor has stopped logging, which presumably means a lot of LSP messages flying around, that mostly end up no-oping in the Razor extension code.
Uninformed opinion: In VS it makes sense for Roslyn to always want the latest dynamic document info, but in VS Code it might be worth Roslyn caching this info, as it really only gets back a single filename.
- By far the majority of time the project spinner is going is after the C# and Razor logs have calmed down, with just the odd file watching related message coming in, so hopefully the trace reveals something useful to someone who knows what they're looking for.

Related to [Bug 1801072: Operation Progress takes a very long time to complete when opening OrchardCore - Boards (visualstudio.com)](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1801072)

**Comment by jasonmalinowski** [comment]
@davidwengier Looking at old bugs quick, is there still any value at looking at this?

**Comment by davidwengier** [comment]
No. IDynamicFile is already not used by default in the C# extension >= v2.88.x so really we're just waiting to be able to delete all of the code :)
